src/mgmt/srv_query.c

```diff
@@ -41,58 +41,60 @@ bs_state_query(void *arg)
 	bio_get_bs_state(&bs_arg->bs_arg_state, bs_arg->bs_arg_type, bxc);
 }
 
+static inline int
+tgt2xs_type(int tgt_id)
+{
+	return tgt_id == BIO_SYS_TGT_ID ? DSS_XS_SYS : DSS_XS_VOS;
+}
+
 /*
  * CaRT RPC handler for management service "get blobstore state" (C API)
  *
  * Internal blobstore states returned for test validation only.
  */
 int ds_mgmt_get_bs_state(uuid_t bs_uuid, int *bs_state)
 {
-	struct smd_dev_info		*dev_info;
-	ABT_thread			 thread;
-	int				 tgt_id;
-	int				 rc;
-	struct bs_state_query_arg	 bs_arg;
+	ABT_thread			thread;
+	int				tgt_id;
+	int				rc;
+	struct bs_state_query_arg	bs_arg;
+	int				xs_type;
 
-	bs_arg.bs_arg_type = SMD_DEV_TYPE_DATA;
 	/*
 	 * Query per-server metadata (SMD) to get target ID(s) for given device.
 	 */
 	if (!uuid_is_null(bs_uuid)) {
-		rc = smd_dev_get_by_id(bs_uuid, &dev_info);
-		if (rc != 0) {
-			D_ERROR("Blobstore UUID:"DF_UUID" not found\n",
+		tgt_id = bio_dev_owner_xs_id(bs_uuid);
+		if (tgt_id < 0) {
+			D_ERROR("Device UUID:"DF_UUID" not found\n",
 				DP_UUID(bs_uuid));
-			return rc;
+			return tgt_id;
 		}
-		if (dev_info->sdi_tgts == NULL) {
-			D_ERROR("No targets mapped to device\n");
-			rc = -DER_NONEXIST;
-			goto out;
-		}
-		/* Default tgt_id is the first mapped tgt */
-		tgt_id = dev_info->sdi_tgts[0];
 	} else {
 		D_ERROR("Blobstore UUID is not provided for state query\n");
 		return -DER_INVAL;
 	}
 
+	xs_type = tgt2xs_type(tgt_id);
+	if (xs_type == DSS_XS_SYS)
+		bs_arg.bs_arg_type = SMD_DEV_TYPE_META;
```

Comment:
What if SCM is configured (not md-on-ssd)? And I don't know why we need device type at all, since the device ID is already provided by user.